pkg/local/diff.go

```diff
@@ -194,14 +197,29 @@ func DiffLocalWithHead(left <-chan apigen.ObjectStats, rightPath string) (Change
 		currentRemoteFile apigen.ObjectStats
 		hasMore           bool
 	)
-	err := filepath.Walk(rightPath, func(path string, info fs.FileInfo, err error) error {
+	absPath, err := filepath.Abs(rightPath)
+	if err != nil {
+		return nil, err
+	}
+	uri := url.URL{Scheme: "local", Path: absPath}
+	adapter, err := local.NewAdapter(absPath, local.WithRemoveEmptyDir(false), local.WithAllowedExternalPrefixes([]string{"/"}))
+	if err != nil {
+		return nil, err
+	}
+	reader, err := adapter.GetWalker(&uri)
+	if err != nil {
+		return nil, err
+	}
+	err = reader.Walk(context.Background(), &uri, block.WalkOptions{}, func(e block.ObjectStoreEntry) error {
+		info, err := os.Stat(e.FullKey)
 		if err != nil {
 			return err
 		}
-		if info.IsDir() || diffShouldIgnore(info.Name()) {
+		if info.IsDir() || diffShouldIgnore(info.Name()) || strings.HasPrefix(e.RelativeKey, local.CacheDirName) {
```

Comment:
We don't actually use the cache so we can just pass an empty path and remove this